arithmetization/src/main/java/net/consensys/linea/sequencer/txselection/selectors/TraceLineLimitTransactionSelector.java

```diff
@@ -151,6 +171,17 @@ public BlockAwareOperationTracer getOperationTracer() {
     return zkTracer;
   }
 
+  private void rememberOverLineCountLimitTransaction(final Transaction transaction) {
+    if (overLineCountLimitCache.size() > overLimitCacheSize) {
+      overLineCountLimitCache.iterator().remove();
```

Comment:
Why not `overLineCountLimitCache.size() >= overLimitCacheSize`, otherwise, it implies the `overLineCountLimitCache.size()` would not always be under or equal to `overLimitCacheSize`

Reply:
Also I see here's the only place we calls remove entry from `overLineCountLimitCache`, why we're not doing any removal in `onTransactionNotSelected` and `onTransactionSelected`?

Reply:
Oh, yes, they shouldn't be removed, we need to keep them in memory to avoid the sync scripts for resubmitting those txns

Reply:
Will fix the size check to be >≠.

About the second point, the behavior is what we need here. Because a bad tx in the cache will never be select, and there is no reason to remove in ‘onTransactionSelected’, and we need to remember it for the future blocks, so we cannot remove it in ’onTrandactionNotSelected’.

So we only remove a bad tx when the cache is full.

Reply:
Yes, @fab-10, I understood after I read it more carefully, that's why I marked as off topic, thanks for the explain anyway